openapi/media_type.go

```diff
@@ -8,20 +8,31 @@ type MediaType struct {
 	Schema *Schema
 }
 
-func (m *MediaType) GetName() string {
-	return m.parent.GetName() + m.name
-}
+//func (m *MediaType) GetName() string {
+// TODO where is this used?
+//
+//	return m.parent.GetName() + "[" + m.node.name + "]"
+//}
 
-func (m *MediaType) getRef() string {
-	return ""
-}
+//func (m *MediaType) getRef() string {
+// FIXME node composable should override this
+//if m.Schema != nil {
+//	return m.Schema.Ref
+//}
+//fmt.Println("borked in media type getref")
+//return ""
+//}
 
 func (m *MediaType) getChildren() map[string]Traversable {
 	return map[string]Traversable{
 		"Model": m.Schema,
 	}
 }
 
-func (m *MediaType) setChild(i string, t Traversable) {
-	m.Schema = t.(*Schema)
+func (m *MediaType) setChild(_ string, t Traversable) {
+	if s, ok := t.(*Schema); ok {
+		m.Schema = s
+		return
+	}
+	panic("(m *MediaType) setChild borked")
```

Comment:
Non blocking: 

Thanks for handling these. I'm on the fence about whether we should be returning and handling errors in the setChild method so that we can more clearly communicate when a user has provided an invalid spec.

Something for an ADR perhaps.